File: src/test/scala/com/exasol/cloudetl/source/AvroSourceTest.scala
package com.exasol.cloudetl.source

import java.nio.file.Paths

class AvroSourceTest extends AbstractSourceTest {

  override val format: String = "avro"

  test("stream returns count of records from AVRO files") {
    val filePath = Paths.get(s"$resourceDir/sales1*.avro")
    assert(getRecordsCount(filePath) === 1998)
  }

  test("stream throws if it cannot create AVRO reader") {
    val nonPath = new org.apache.hadoop.fs.Path(s"$resourceDir/notFile.avro")
    val thrown = intercept[SourceValidationException] {
      getSource(nonPath).stream().size
    }
    assert(thrown.getMessage().startsWith("E-CSE-26"))
    assert(thrown.getMessage().contains(s"Could not create Avro reader for path '$nonPath'."))
  }

}

File: src/test/java/com/exasol/cloudetl/source/AvroSourceTest.java
package com.exasol.cloudetl.source;

import static org.junit.jupiter.api.Assertions.*;

import java.nio.file.Paths;

import org.apache.hadoop.fs.Path;
import org.junit.jupiter.api.Test;

class AvroSourceTest extends AbstractSourceTest {
    @Override
    protected String getFormat() {
        return "avro";
    }

    @Test
    void streamReturnsCountOfRecordsFromAvroFiles() throws java.io.IOException {
        assertEquals(1998, getRecordsCount(Paths.get(this.resourceDir + "/sales1*.avro")));
    }

    @Test
    void streamThrowsIfItCannotCreateAvroReader() {
        final Path nonPath = new Path(this.resourceDir + "/notFile.avro");
        final SourceValidationException thrown = assertThrows(SourceValidationException.class,
                () -> getSource(nonPath).stream().size());
        assertTrue(thrown.getMessage().startsWith("E-CSE-26"));
        assertTrue(thrown.getMessage().contains("Could not create Avro reader for path '" + nonPath + "'."));
    }
}
